### PR TITLE
New test cases

### PR DESCRIPTION
**Pull request by LiviuDanu**

### CODE REVIEW

File: serenity.properties [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+#serenity.browser.width = 1450
+#serenity.browser.height = 900
+#serenity.use.unique.browser = true
+#serenity.browser.maximize = true
+serenity.restart.browser.for.each = feature
+
+# Environment
+########################
+#webdriver.base.url = https://vrv.co
```
Comment (pavel0712) [inline serenity.properties]: fix URL string

File: serenity.properties [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+webdriver.base.url = https://stage.crunchyroll.com
+#webdriver.base.url = https://cx-proto0.com
```
Comment (pavel0712) [inline serenity.properties]: fix URL string

File: src/main/java/pages/LogInPage.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    @FindBy(id = "signup_with_profile_opt_in_to_marketing")
+    public WebElementFacade subscribeCheckBox;
+
+    @FindBy(id = "signup_submit_button")
+    public WebElementFacade submitButton;
+
+    @FindBy(css = "output.input-error-message")
+    public WebElementFacade emailErrorlabel;
+
+
```
Comment (pavel0712) [inline src/main/java/pages/LogInPage.java]: Remove extra empty lines

File: src/main/java/pages/HomePage.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    public WebElementFacade privacyPolicyButton;
+
+    public void clickOnLoginButton() {
+        loginButton.click();
+    }
+
+    public void openHomePage() {
+        navigateToPath("");
+    }
+
```
Comment (pavel0712) [inline src/main/java/pages/HomePage.java]: Remove extra empty lines

File: src/main/java/pages/HomePage.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+public class HomePage extends AbstractPage {
+
+    @FindBy(css = ".login .header-icon")
+    public WebElementFacade loginButton;
+    @FindBy(id = "header_search_input")
+    public WebElementFacade searchBox;
+    @FindBy(id = "header_search_submit_beta")
+    public WebElementFacade searchButton;
+    @FindBy(xpath = "//*[@id=\"footer_menu\"]/div[1]/div[6]/ul/li[4]/a")
```
Comment (pavel0712) [inline src/main/java/pages/HomePage.java]: Could we find better selector?

File: src/main/java/components/Header.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import pages.AbstractPage;
+
+public class Header extends AbstractPage{
+    @FindBy(id = "logo_beta")
+    public WebElementFacade logo;
+    @FindBy(css = ".login .header-icon")
+    public WebElementFacade logInButton;
+    @FindBy(id = "header_search_input")
+    public WebElementFacade searchField;
+    @FindBy(xpath = "//*[@id=\"header_menubar_beta\"]/ul/li[1]/a")
```
Comment (pavel0712) [inline src/main/java/components/Header.java]: Could we find better selector

File: src/main/java/steps/HomeSteps.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+import static org.hamcrest.Matchers.equalToIgnoringCase;
+
+public class HomeSteps  extends BaseSteps {
+
+    HomePage homePage;
+    Header header;
+
+    @Step("Click on 'Login' button")
+    public void clickOnLoginButton() {
+        homePage.clickOnLoginButton();
```
Comment (pavel0712) [inline src/main/java/steps/HomeSteps.java]: click on button directly, do not create method for it

File: src/main/java/steps/HomeSteps.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+    @Step("Open Home Page")
+    public void openHomePage() {
+        homePage.openHomePage();
+    }
+
+    @Step("Search For Naruto")
+    public void searchFor(String src){
+        homePage.searchBox.sendKeys(src);
+        //homePage.searchButton.click();
```
Comment (pavel0712) [inline src/main/java/steps/HomeSteps.java]: remove dead code

File: src/main/java/steps/HomeSteps.java [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    @Step
+    public void shouldSeeHomePage(){
+        assertThat("HomePage is not Displayed", getDriver().getCurrentUrl(), is(equalToIgnoringCase("http://stage.crunchyroll.com/")));
+    }
+    @Step
+    public void clickOnMangaButton(){
+        header.mangaButton.click();
+    }
+
+
```
Comment (pavel0712) [inline src/main/java/steps/HomeSteps.java]: empty line

File: src/main/java/steps/LogInSteps.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        logInPage.genderRadioButton.click();
+    }
+    @Step("Unsubscribe")
+    public void unsubscribe(){
+        logInPage.subscribeCheckBox.click();
+    }
+    @Step("Click Register Button")
+    public void clickRegisterButton(){
+        logInPage.submitButton.click();
+        waitABit(10000);
```
Comment (pavel0712) [inline src/main/java/steps/LogInSteps.java]: Why we using waitABit()?

File: src/main/java/steps/LogInSteps.java [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    @Step
+    public void shouldSeeAnErrorMessage(String email) {
+        assertThat("The email error message is incorrect", logInPage.emailErrorlabel.getText(), is(equalToIgnoringCase(email)));
+    }
+
+
```
Comment (pavel0712) [inline src/main/java/steps/LogInSteps.java]: empty line

File: src/main/java/steps/MangaSteps.java [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+import net.thucydides.core.annotations.Step;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.text.IsEqualIgnoringCase.equalToIgnoringCase;
+
+public class MangaSteps extends BaseSteps{
+    @Step
+    public void shouldSeeMangaPage(){
+        assertThat("Manga Page Is not displayed", getDriver().getCurrentUrl(), is(equalToIgnoringCase("http://stage.crunchyroll.com/comics/manga")));
```
Comment (pavel0712) [inline src/main/java/steps/MangaSteps.java]: Add for MangaPage properti : URL = "/comics/manga", and concatenate with BaseURL

File: src/main/java/steps/SearchResultsSteps.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+    SearchResultsPage searchResultsPage;
+
+
+    @Step("Checking if the third result contains naruto")
+    public void shouldSeeExpectedResults(String word){
+            assertThat("The result does not contain Naruto", searchResultsPage.containsNaruto(word), is(true));
+        }
+    }
+
+
```
Comment (pavel0712) [inline src/main/java/steps/SearchResultsSteps.java]: empty lines

File: src/main/java/steps/ShowsSteps.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+    ShowsPage showsPage;
+    Header header;
+
+    @Step
+    public void clickOnLogo(){
+        header.logo.click();
+    }
+    @Step
+    public void shoulSeeShowsPage(){
+        assertThat("Shows Page is not displayed", getDriver().getCurrentUrl(), is(equalToIgnoringCase("http://stage.crunchyroll.com/videos/anime")));
```
Comment (pavel0712) [inline src/main/java/steps/ShowsSteps.java]: Extract path for ShowPage as property

File: src/test/java/RegisterTest.java [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+import org.junit.runner.RunWith;
+
+@Story(Application.Register.SuccessRegister.class)
+@RunWith(SerenityRunner.class)
+@WithTag("register")
+public class RegisterTest extends BaseTest {
+
+    //region 08/08 tests
+
+/*@Test
```
Comment (pavel0712) [inline src/test/java/RegisterTest.java]: Why dead code is here?

File: src/main/java/pages/AbstractPage.java [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+public class AbstractPage extends PageObject {
+
+    public void navigateURL(String url) {
+        getDriver().get(url);
+    }
+
+    public void navigateToPath(String path) {
+        navigateURL(Constants.BASE_URL + path);
+    }
+    public void send(Keys key) {
```
Comment (pisarenco92) [inline src/main/java/pages/AbstractPage.java]: change the method name (ex: sendKey)

File: src/main/java/pages/SearchResultsPage.java [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+
+public class SearchResultsPage extends AbstractPage {
+
+    @FindBy(xpath = "//*[@id=\"aux_results\"]/ul//li/a/span[2]")
+    public List<WebElementFacade> results;
+
+    public boolean containsNaruto(String word){
```
Comment (pisarenco92) [inline src/main/java/pages/SearchResultsPage.java]: change the name of the method to have general meaning

File: src/main/java/steps/HomeSteps.java [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    @Step("Scrolling Down")
+    public void scrollDown(){
+        homePage.scrollDown();
+    }
+    @Step("Click privacy policy link")
+    public void clickPrivacyPolicyLink(){
+        homePage.privacyPolicyButton.click();
+    }
+    @Step
```
Comment (pisarenco92) [inline src/main/java/steps/HomeSteps.java]: Add text in Step

File: src/main/java/steps/HomeSteps.java [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+    public void shouldSeeLogo(){
+        assertThat("Logo is not displayed", header.logo.isVisible());
+    }
+    @Step
```
Comment (pisarenco92) [inline src/main/java/steps/HomeSteps.java]: Add text in Step

File: src/main/java/steps/HomeSteps.java [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+    public void shouldSeeLogInButton(){
+        assertThat("Log In Button Is Not Displayed", header.logInButton.isVisible());
+    }
+    @Step
```
Comment (pisarenco92) [inline src/main/java/steps/HomeSteps.java]: Add text in Step

File: src/main/java/steps/HomeSteps.java [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+    public void shouldSeeSearchField(){
+        assertThat("Search Field Is Not Visible", header.searchField.isVisible());
+    }
+    @Step
```
Comment (pisarenco92) [inline src/main/java/steps/HomeSteps.java]: Add text in Step

File: src/main/java/steps/HomeSteps.java [hunk 22]
```diff
@@ -0,0 +1,72 @@
...
+    public void clickOnSwowsButton(){
+        header.showsButton.click();
+    }
+    @Step
```
Comment (pisarenco92) [inline src/main/java/steps/HomeSteps.java]: Add text in Step

File: src/main/java/steps/HomeSteps.java [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+    public void shouldSeeHomePage(){
+        assertThat("HomePage is not Displayed", getDriver().getCurrentUrl(), is(equalToIgnoringCase("http://stage.crunchyroll.com/")));
+    }
+    @Step
```
Comment (pisarenco92) [inline src/main/java/steps/HomeSteps.java]: Add text in Step

File: src/main/java/steps/LogInSteps.java [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+import static org.hamcrest.Matchers.*;
+import static org.hamcrest.MatcherAssert.*;
+
+public class LogInSteps extends BaseSteps {
+
+    LogInPage logInPage;
+
+    @Step("Fill The Email Text Field")
+    public void fillEmailTextField(String email){
+        logInPage.emailTextField.click();
```
Comment (pisarenco92) [inline src/main/java/steps/LogInSteps.java]: why not to have a method in AbstractPage to enter text in fields

File: src/main/java/steps/LogInSteps.java [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+    @Step("Unsubscribe")
+    public void unsubscribe(){
+        logInPage.subscribeCheckBox.click();
+    }
+    @Step("Click Register Button")
+    public void clickRegisterButton(){
+        logInPage.submitButton.click();
+        waitABit(10000);
+    }
+    @Step
```
Comment (pisarenco92) [inline src/main/java/steps/LogInSteps.java]: Add text in Step

File: src/main/java/steps/MangaSteps.java [hunk 26]
```diff
@@ -0,0 +1,14 @@
+package steps;
+
+import net.thucydides.core.annotations.Step;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.text.IsEqualIgnoringCase.equalToIgnoringCase;
+
+public class MangaSteps extends BaseSteps{
+    @Step
```
Comment (pisarenco92) [inline src/main/java/steps/MangaSteps.java]: Add text in Step

File: src/main/java/steps/ShowsSteps.java [hunk 27]
```diff
@@ -0,0 +1,24 @@
...
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.text.IsEqualIgnoringCase.equalToIgnoringCase;
+
+public class ShowsSteps extends BaseSteps {
+    ShowsPage showsPage;
+    Header header;
+
+    @Step
```
Comment (pisarenco92) [inline src/main/java/steps/ShowsSteps.java]: Add text in Step

File: src/test/java/HeaderTest/HeaderTests.java [hunk 28]
```diff
@@ -0,0 +1,51 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@Story(Application.Register.SuccessRegister.class)
+@RunWith(SerenityRunner.class)
+@WithTag("register")
+
+public class HeaderTests extends BaseTest {
+    @Test
+    @Title("crunchyroll.com")
```
Comment (pisarenco92) [inline src/test/java/HeaderTest/HeaderTests.java]: Add test case ID

File: src/test/java/HeaderTest/HeaderTests.java [hunk 29]
```diff
@@ -0,0 +1,51 @@
...
+    public void logoLoginButtonSearchFieldVisibilityTest(){
+        homeSteps.openHomePage();
+        homeSteps.shouldSeeLogo();
+        homeSteps.shouldSeeLogInButton();
+        homeSteps.shouldSeeSearchField();
+
+    }
+
+    @Test
+    @Title("Crunchyroll Home")
```
Comment (pisarenco92) [inline src/test/java/HeaderTest/HeaderTests.java]: Add test case ID

File: src/test/java/HeaderTest/HeaderTests.java [hunk 30]
```diff
@@ -0,0 +1,51 @@
...
+    public void crunchyrollHomeTest(){
+        homeSteps.openHomePage();
+        homeSteps.clickOnSwowsButton();
+        showsSteps.clickOnLogo();
+        homeSteps.shouldSeeHomePage();
+    }
+    @Test
+    @Title("Shows Test")
```
Comment (pisarenco92) [inline src/test/java/HeaderTest/HeaderTests.java]: Add test case ID

File: src/test/java/HeaderTest/HeaderTests.java [hunk 31]
```diff
@@ -0,0 +1,51 @@
...
+    public void showsTest(){
+        homeSteps.openHomePage();
+        homeSteps.clickOnSwowsButton();
+        showsSteps.shoulSeeShowsPage();
+    }
+    @Test
+    @Title("Manga Test")
```
Comment (pisarenco92) [inline src/test/java/HeaderTest/HeaderTests.java]: Add test case ID

**Pull request closed without merging**